packages/zoltan2/src/algorithms/partition/Zoltan2_AlgPuLP.hpp

```diff
@@ -492,33 +535,50 @@ void AlgPuLP<Adapter>::partition(
   pe = pl.getEntryPtr("pulp_seed");
   if (pe) pulp_seed = pe->getValue(&pulp_seed);
 
-  // Create PuLP's partitioning data structure
-  pulp_part_control_t ppc = {vert_imbalance, edge_imbalance,
-    do_lp_init, do_bfs_init, do_repart,
-    do_edge_bal, do_maxcut_min,
-    verbose_output, pulp_seed};
-
 
   if (verbose_output) {
-    printf("procid: %d, n: %i, m: %li, vb: %f, eb: %f, p: %i\n",
+    printf("procid: %d, n: %i, m: %li, vb: %f, eb: %f, imb: %f, p: %i\n",
       problemComm->getRank(), 
-      num_verts, num_edges, vert_imbalance, edge_imbalance, num_parts);
+      num_verts, num_edges, 
+      vert_imbalance, edge_imbalance, imbalance, num_parts);
   }
 
   // Call partitioning; result returned in parts array
 #ifndef HAVE_ZOLTAN2_MPI
+  // Create PuLP's partitioning data structure
+  pulp_part_control_t ppc = {vert_imbalance, edge_imbalance,
+    do_lp_init, do_bfs_init, do_repart,
+    do_edge_bal, do_maxcut_min,
+    verbose_output, pulp_seed};
+
   ierr = pulp_run(&g, &ppc, parts, num_parts);  
 
   env->globalInputAssertion(__FILE__, __LINE__, "pulp_run", 
     !ierr, BASIC_ASSERTION, problemComm);
 #else
+  // Create XtraPuLP's partitioning data structure
+  double* constraints = NULL;
```

Comment:
Prefer `nullptr` to `NULL`.  Also, use of raw `new` and `delete` is an antipattern.